src/NaadSocketConnection.php

```diff
@@ -0,0 +1,126 @@
+<?php
+namespace Bcgov\NaadConnector;
+
+use Bcgov\NaadConnector\Database;
+use Monolog\Logger;
+
+/**
+ * NaadSocketConnection class connects to the NAAD socket.
+ *
+ * @category Client
+ * @package  NaadConnector
+ * @author   Michael Haswell <Michael.Haswell@gov.bc.ca>
+ * @license  https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link     https://alerts.pelmorex.com/
+ */
+class NaadSocketConnection
+{
+
+    protected static int $MAX_MESSAGE_SIZE = 5000000;
+
+    protected string $name;
+
+    protected string $address;
+
+    protected int $port;
+
+    protected Logger $logger;
+
+    protected DestinationClient $destinationClient;
+
+    /**
+     * Constructor for NaadClient.
+     *
+     * @param string            $name              The name of the NAAD connection
+     *                                             instance.
+     * @param string            $socketUrl         The URL of the NAAD socket to
+     *                                             connect to.
+     * @param DestinationClient $destinationClient An instance of DestinationClient.
+     * @param Logger            $logger            An instance of Monolog/Logger.
+     * @param integer           $port              The port of the NAAD socket to
+     *                                             connect to.
+     */
+    public function __construct(
+        string $name,
+        string $socketUrl,
+        DestinationClient $destinationClient,
+        Logger $logger,
+        int $port = 8080,
+    ) {
+        $this->name              = $name;
+        $this->address           = $socketUrl;
+        $this->destinationClient = $destinationClient;
+        $this->logger            = $logger;
+        $this->port              = $port;
+    }
+
+    /**
+     * Connects to the NAAD socket at the given URL and listens.
+     *
+     * @return int An exit code.
+     */
+    public function connect(): int
+    {
+        // Create a TCP/IP socket.
+        $this->logger->info('Creating socket');
+        $socket = socket_create(AF_INET, SOCK_STREAM, SOL_TCP);
+        if ($socket === false ) {
+            $this->logger->error(
+                'socket_create() failed: reason: {error}',
+                [
+                    'error' => socket_strerror(socket_last_error()),
+                ]
+            );
+            return 2;
+        } else {
+            $this->logger->info('OK.');
```

Comment:
not a really helpful message... maybe 
```suggestion
            $this->logger->info('Socket OK.');
```

Reply:
I've updated the log messages in this file to be more descriptive.